Update azure-stack-register.md to include resource provider registration

BradleyBartlett opened the pull request:
This update is in response to several internal issues with the steps involving registering a resource provider being unclear and causing the script to fail.

azurecla commented:
Hi __@v-kagri__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (v-kagri). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

Commit:
Update azure-stack-register.md

This update is in response to several internal issues with the steps involving registering a resource provider being unclear and causing the script to fail.
